### ISSUE TITLE
Autofill from password manager not working in Mobile Safari

### ISSUE THREAD

**Issue by michaeldyrynda** [livewire/livewire#1081]
### Description

Per issue title

### Steps to reproduce

I am using the [TALL](https://github.com/laravel-frontend-presets/tall) Laravel frontend preset, and have found that the login form (and possibly others, given the nature of the issue) don't appear to work correctly when using a password manager - 1Password in my instance - for autofill.

My component properties are typed i.e. `public string $email`, however, I get a 500 error when using the password manager, that the `$email` property cannot be `null`. I can make the `$email` property optional (`?string`), but this merely allows the component validation to run, which will say that the email field is required.

Otherwise typing my email and password, and even manually pasting the password into the input works correctly.

Video of errant behaviour: http://dyry.me/6d7sgj

**Context**
- Livewire version: 1.1.0
- Laravel version: 7.14.1
- Browser: Mobile Safari (iOS 13.5)

**Comment by nuernbergerA** [comment]
I assume its a problem with the event that's triggered by 1Password after filled in each field, guess it's `keypress` with an empty/0 `key`/`keyCode`.

Sadly `monitorEvents()` is only available on Chrome.

@michaeldyrynda are you using `wire:model` (input) or `wire:model.lazy` (change) or something else?

As a quick test, you could use [alpineJS](https://laravel-livewire.com/docs/alpine-js) to catch the event and forward it to livewire.
```blade
<input type="password" wire:model="password" x-data @what-ever-event-1Password-emits="$dispatch('input', @this.value)" />
```

---
Maybe replates to #1034

**Comment by nuernbergerA** [comment]
I found a nice [code snippet](https://stackoverflow.com/questions/7439570/how-do-you-log-all-events-fired-by-an-element-in-jquery#answer-57837123) to debug events in pure js (should work in all browsers).

**Comment by nuernbergerA** [comment]
Ok @michaeldyrynda you could use https://nuernbergera.github.io/debug-events/login/ to store login and https://nuernbergera.github.io/debug-events/ to test your login with your password manager

**Be aware: Password will be shown in cleartext**

**Comment by acurrieclark** [comment]
I am finding this issue too in Livewire v1.3.1. Appears to be the same issue as #943 which is marked as fixed in v1.1.

I am only currently seeing the issue in mobile Safari.

**Comment by vluijkx** [comment]
This was solved for us in version 1.3.0 but then reappeared in 1.3.1. Both on desktop Safari and mobile Safari.

**Comment by stefro** [comment]
I can confirm what @vluijkx noted. It was gone in 1.3.0 and reappeared in 1.3.1

**Comment by ryangjchandler** [comment]
@acurrieclark @vluijkx @stefro The problem in the OP is that Livewire isn't picking up on the password autofill at all, none of the regular change / input events are being fired which is a common problem across all frameworks since each browser & vendor will implement them slightly differently.

**Comment by acurrieclark** [comment]
Ah, so it is. I confess I hadn't watched the video.

That being said, I have only had autofill issues with Safari, and for some reason or another they disappeared with 1.3.0 and returned in 1.3.1. Perhaps a separate issue, but likely relevant.

**Comment by ryangjchandler** [comment]
@acurrieclark there was a small change from 1.3.0 to 1.3.1 regarding event listeners - I'll have to look into that one. The problem with Safari, especially on iOS, is that password managers don't interact directly with the browser. Instead, they go through the iOS Autofill API so it's not really down to them to fire all of the correct events on the input, it's more down to Apple.

**Comment by stefro** [comment]
Thanks @ryangjchandler for looking into this! I hope you can find the problem!

**Comment by ryangjchandler** [comment]
@stefro theres not necessarily a problem, it's more of a quirk so the "fix" is more of a hack, haha. 

I'll get a PR in for this soon, testing it is a bit of a pain in the neck.

**Comment by stefro** [comment]
@ryangjchandler haha. I understand. 
Your efforts are much appreciated!

**Comment by glaesser** [comment]
has there been any update on this, any temporary workaround/fix we can deploy ourselves?

**Comment by acurrieclark** [comment]
@glaesser I am removing all `wire:model` directives from my logging form then and replacing them with the alpine equivalent. I then use alpine to capture the form submit event, fill the livewire models using `@this.set` and then call the login method with `@this.call`.

Works fine for me for now while a proper fix is put in place.

**Comment by stefro** [comment]
@acurrieclark That is a clever solution. But it's still a workaround.
We're using it on a order form with address, country etc fields. Livewire is great to show relevant info as soon as the country field changes. But when a user is on Safari and uses the autofill, everything breaks.

Strangely, not on Livewire 1.3.0, but we cannot stay on this version forever. I really wish I could help out to make a PR for this, but I'm not an expert on frontend.

**Comment by redbastie** [comment]
PR that fixes the bug with IE11 support: https://github.com/livewire/livewire/pull/1696

**Comment by calebporzio** [comment]
Safari autofill should be fixed in 2.2.8

Please run `php artisan view:clear`, update Livewire, and report here if the problem is fixed, otherwise, re-submit please.

Thanks!

**Comment by catasuisse** [comment]
It still not seems to work with Face-ID:

https://www.youtube.com/watch?v=E2GkWf6_sCg

**Comment by zain** [comment]
> Safari autofill should be fixed in 2.2.8
> 
> Please run `php artisan view:clear`, update Livewire, and report here if the problem is fixed, otherwise, re-submit please.
> 
> Thanks!

Could you point us to the commit that fixed it?

**Comment by Neeraj1005** [comment]
while loading the page my email and password field automatically fills in chrome... But in Mozilla or Brave it working fine.
